### PR TITLE
[java-micronaut] Allow default implementation to return HTTP 501

### PR DESCRIPTION
**Pull request by auke-**
The default controller implementation returns an empty response. While
this might be an expected default for some users we want the default
implementation to return HTTP 501 Not Implemented instead. This will
support rapid development when the specs aren't finalized yet and thus
change often while preventing the methods from being unimplemented by
accident without the user of the API noticing it.

By adding a property which allows the generation of default methods
which return HTTP 501 developers can opt-in for this behavior.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

@andriy-dmytruk

### CODE REVIEW

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaMicronautServerCodegen.java [hunk 1]
```diff
@@ -20,6 +20,7 @@ public class JavaMicronautServerCodegen extends JavaMicronautAbstractCodegen {
     public static final String OPT_CONTROLLER_PACKAGE = "controllerPackage";
     public static final String OPT_GENERATE_CONTROLLER_FROM_EXAMPLES = "generateControllerFromExamples";
     public static final String OPT_GENERATE_CONTROLLER_AS_ABSTRACT = "generateControllerAsAbstract";
+    public static final String OPT_GENERATE_CONTROLLER_NOT_IMPLEMENTED = "generateControllerNotImplemented";
```
Comment (andriy-dmytruk) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaMicronautServerCodegen.java:23]: Can we come up with a more descriptive name for the option. 
Maybe `operationsReturnNotImplementedByDefault` or `generateOperationsToReturnNotImplemented`?

**Review by andriy-dmytruk — CHANGES_REQUESTED** [review]
@auke- Thank you for the PR.
Looks like a valuable contribution to me.

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaMicronautServerCodegen.java (hunk 1 above)
Comment (auke-) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaMicronautServerCodegen.java]: Done!

**Review by andriy-dmytruk — APPROVED** [review]
Tested locally - works as expected. Thanks.
@wing328 Can you merge?

**Comment by wing328** [comment]
Thanks again for the PR, which has been included in the v6.0.0 release: https://twitter.com/oas_generator/status/1529669733934841856

**Pull request merged**